packages/gdl-frontend/lib/offlineLibrary.js

```diff
@@ -0,0 +1,264 @@
+// @flow
+import { coverImageUrl } from 'gdl-image';
+import localForage from 'localforage';
+
+import type { BookDetails, Chapter } from '../types';
+import { fetchChapter } from '../fetch';
+
+const CACHE_NAME = 'gdl-offline';
+// 7 days
+const MAX_AGE_MS = 7 * 24 * 60 * 60 * 1000;
+
+function openCache() {
+  return window.caches.open(CACHE_NAME);
+}
+
+/**
+ * We keep track of when the book was offlined, so we can expire it after a certain time has passed
+ */
+export class TimestampModel {
```

Comment:
I feel that this file has too much content. What do you think about extracting the classes to different files? Or maybe create an offline folder with the different classes and utils function for easer readability?